app/controllers/api/v1/owners_controller.rb

```diff
@@ -1,7 +1,7 @@
 class Api::V1::OwnersController < Api::BaseController
-  before_action :authenticate_with_api_key, except: %i[show gems]
+  before_action :find_api_key, only: %i[create destroy]
   before_action :find_rubygem, except: :gems
-  before_action :verify_gem_ownership, except: %i[show gems]
+  before_action :verify_gem_ownership, only: %i[create destroy]
```

Comment:
As above, prefer `except` so the default for new actions is to be more secure. Better that newly added actions fail safe than fail insecurely.